### PR TITLE
Explicitly activate/deactivate extension for cleaner integration test runs

### PR DESCRIPTION
**Pull request by plemarquand**
The `globalWorkspaceContextPromise` would activate the extension once and then always return the same workspace context for every test run. Its a limitation that we cannot actually activate the extension multiple times in tests (there is no extension deactivate API offered by VS Code), however we can dispose and recreate the WorkspaceContext at will to ensure a cleaner test environment.

`activateExtension` will call `extension.activate()` the first time it's used, and then `deactivateExtension` will dispose of the returned WorkspaceContext. Subsequent calls to `activateExtension` will use the activation method returned by the Swift extension's entrypoint API to recreate the `WorkspaceContext`. If `activateExtension` is called again without a matching call to `deactivateExtension` an error is thrown.

We make sure there are no running tasks up to 10 seconds after a deactivation is requested. Long running tasks should be cleaned at the end of a test as they can interfere with other tests.

Logs sent to the `SwiftOutputChannel` are captured and only printed if a test fails. This behaviour is only enabled if the `VSCODE_TEST` environment variable is set, and is configured as enabled in CI.

### CODE REVIEW

**Comment by plemarquand** [comment]
@swift-server-bot test this please

**Comment by michael-weng** [comment]
LGTM, but after you rebase you might have to fix up the other integration tests that got merged. :P

**Review by matthewbastien — APPROVED** [review]

File: src/TestExplorer/TestExplorer.ts [hunk 1]
```diff
@@ -110,6 +112,9 @@ export class TestExplorer {
     }
 
     dispose() {
+        this.controller.refreshHandler = undefined;
+        this.controller.resolveHandler = undefined;
```
[the remark this answers is not in the document]
Comment (plemarquand) [reply src/TestExplorer/TestExplorer.ts:116]: not quite, one is the refresh handler, one is the resolve handler ;)

[the remark this answers is not in the document]
Comment (matthewbastien) [reply src/TestExplorer/TestExplorer.ts:116]: Ha! Seems like I can't read properly.

**Pull request merged**